src/math/p_tan.c

```diff
@@ -31,38 +31,38 @@ static inline float __p_tan_pi_4(const float x)
 }
 
 /*
- * pi/4 <= x <= pi/2
+ * 0 <= x <= pi/2
  * x = x' + pi/4
  * tan x = tan(x' + pi/4) = (tan x' + 1) / (1 - tan x')
  */
 static inline float __p_tan_pi_2(const float x)
 {
     float x_, tanx_;
+    if (x <= pi_4)
```

Comment:
On platforms with `math.h`, `M_PI_4` and other constants are defined with type `double`. When you mix single and double floating-point arithmetic the compiler produces much slower code.